### PR TITLE
CORE: fix timeout handle for triggered post

### PR DESCRIPTION
**Pull request by Sergei-Lebedev**
## What
Correctly handle time out error for triggered task
* For TL NCCL need to use commAbort, otherwise commDestroy could hang
* For other TLs need to destroy executor during finalize

### CODE REVIEW

**Comment by Sergei-Lebedev** [comment]
cc @Fuzzkatt

**Review by manjugv — APPROVED** [review]

**Review by bureddy — APPROVED** [review]

**Comment by Fuzzkatt** [comment]
Fixes the following unit tests on upstream Pytorch CI:
* test_verify_model_across_rank_with_logger: https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/distributed/distributed_test.py#L7722
* test_verify_model_across_rank_without_logger:  https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/distributed/distributed_test.py#L7729
* test_ddp_model_diff_shape_across_ranks: https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/distributed/distributed_test.py#L7735
* test_ddp_model_diff_num_params_across_ranks: https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/distributed/distributed_test.py#L7776

In general, when passing in mismatched inputs across ranks, UCC would originally fail to catch the error within the function itself and propagate the error into a barrier timeout. With this fix, UCC catches the error within the function correctly.

File: src/components/tl/nccl/tl_nccl.h [hunk 1]
```diff
@@ -82,6 +82,7 @@ UCC_CLASS_DECLARE(ucc_tl_nccl_context_t, const ucc_base_context_params_t *,
 
 typedef struct ucc_tl_nccl_team {
     ucc_tl_team_t        super;
+    ucc_status_t         comm_state;
```
Comment (samnordmann) [inline src/components/tl/nccl/tl_nccl.h:85]: update copyright?

File: src/core/ucc_coll.c [hunk 2]
```diff
@@ -368,21 +351,16 @@ static ucc_status_t ucc_trigger_complete(ucc_coll_task_t *parent_task,
 
     if (!(task->flags & UCC_COLL_TASK_FLAG_EXECUTOR)) {
         task->executor = parent_task->executor;
+        task->flags |= (UCC_COLL_TASK_FLAG_EXECUTOR_STOP |
```
Comment (samnordmann) [inline src/core/ucc_coll.c:354]: update copyright ?

**Review by samnordmann — APPROVED** [review]

**Pull request merged**